src/Spice86.Core/Emulator/InterruptHandlers/Common/Callback/EmulationLoopRecalls.cs

```diff
@@ -29,18 +34,18 @@ public EmulationLoopRecalls(InterruptVectorTable ivt, State state, Stack stack,
     /// <returns>Returns the scancode byte.</returns>
     public byte ReadBiosInt16HGetKeyStroke() {
```

Comment:
I think this should be called from Spice86.Core.Emulator.InterruptHandlers.Input.Keyboard.KeyboardInt16Handler.GetKeystroke().
That is the one that should watch the keyboard buffer to be filled by INT 9. (our current implementation is broken since it doesn't wait if the buffer is empty but returns 0x00)
Check out a real bios handling of INT 9 and 16: https://github.com/kaneton/appendix-bios/blob/master/src/keybd.asm 
INT 9: translate keyboard port data to scancode and asciicode for buffer
INT 16: query/read buffer
If we need to just run the CPU for interrupts while waiting for an entry in the keyboard buffer (int 16 00) we can write a small asm routine in our bios memory to do that.

Reply:
Done, although not with ASM.